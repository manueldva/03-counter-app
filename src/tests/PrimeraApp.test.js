import React from 'react';

import '@testing-library/jest-dom';
//import { render  } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow, mount, render } from 'enzyme';

describe('pruebas en <PrimeraApp />', () => {
    
     /*
    test('debe mostrar el msj "hola soy goku" ', () => {
   
       const saludo = 'Hola, Soy Goku';

        const { getByText }  = render( <PrimeraApp saludo={ saludo }  /> )
        
        expect( getByText(saludo ) ).toBeInTheDocument();
    
    });
    */

    test('debe mostrar <PrimeraApp /> correctamente ', () => {
        
        const saludo = 'Hola, Soy Goku';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } /> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar el subtitulo enviado por props ', () => {
        
        const saludo = 'Hola, Soy Goku';
        const subTitulo ='Soy un subtitulo';

        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo } 
                subtitulo={ subTitulo }
            /> 
        );

        const textoParrafo = wrapper.find('p').text();

        expect( textoParrafo ).toBe( subTitulo);
    });
    
    
});
