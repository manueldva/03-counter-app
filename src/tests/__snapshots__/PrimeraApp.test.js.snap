// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
